crates/pgt_workspace/src/workspace/server.rs

```diff
@@ -307,20 +254,26 @@ impl Workspace for WorkspaceServer {
             Some("Statement execution not allowed against database.".into())
         };
 
-        for (stmt, _, txt) in eligible_statements {
-            let title = format!(
-                "Execute Statement: {}...",
-                txt.chars().take(50).collect::<String>()
-            );
-
-            actions.push(CodeAction {
-                title,
-                kind: CodeActionKind::Command(CommandAction {
-                    category: CommandActionCategory::ExecuteStatement(stmt.id),
-                }),
-                disabled_reason: disabled_reason.clone(),
-            });
-        }
+        let actions = parser
+            .iter_with_filter(
+                DefaultMapper,
+                CursorPositionFilter::new(params.cursor_position),
+            )
+            .map(|(stmt, _, txt)| {
+                let title = format!(
+                    "Execute Statement: {}...",
+                    txt.chars().take(50).collect::<String>()
```

Comment:
beautiful

Reply:
also works out of the box with sub statements like sql function bodies and the workspace implementation doesnt even know it